src/redux/reducer.js

```diff
@@ -0,0 +1,51 @@
+import { combineReducers } from 'redux'
+
+import { SEARCH_MOVIES, MOVIES_FETCH_SUCCEEDED, DETAILS_FETCH_SUCCEEDED, ERROR } from './actions'
+
+const searchReducerDefaultState = {results: [], remainingResults: 0, searchString: '', isLoading: false}
+
+const searchReducer = (state = searchReducerDefaultState, action) => {
+  console.log("searchReducer " + action.type);
+  switch (action.type) {
+    case MOVIES_FETCH_SUCCEEDED:
+      //on first page - return new results
+      if (action.payload.page === 1) {
+        return action.payload
+      }
+      //else - add results to existing
+      else {
+        let newState = { ...state, ...action.payload , isLoading: false}
+        newState.results = [...(state.results), ...(action.payload.results)]
+        return newState
+      }
+      break;
+    //on search change to loading
+    case SEARCH_MOVIES:
+      return { ...state, isLoading: true }
+    default:
```

Comment:
Why not add `SHOW_MOVIE_DETAILS` with `isLoading: true` as well?

Reply:
I did add it in the detailsReducer